src/Facebook/FacebookClient.php

```diff
@@ -162,7 +162,11 @@ public function getBaseGraphUrl($postToVideoUrl = false)
      */
     public function prepareRequestMessage(FacebookRequest $request)
     {
-        $postToVideoUrl = $request->containsVideoUploads();
+        $pathElements = explode('/', $request->getUrl());
+        $lastInPath = end($pathElements);
+        $postToVideoUrl = $request->containsVideoUploads() ||
+            ($lastInPath == 'videos' && $request->getMethod() === 'POST');
```

Comment:
Not sure why you added these lines. The `containsVideoUploads()` detects file uploads. We don't want to do any magic guessing what the user is doing based on the endpoint they are poking. :)